src/Sylius/Component/Core/Model/ProductVariant.php

```diff
@@ -173,7 +173,10 @@ public function getPrice()
      */
     public function setPrice($price)
     {
-        $this->price = (int)$price;
+        if (!is_int($price)) {
+            throw new \InvalidArgumentException('Price must be an integer.');
```

Comment:
IMO this is wrong, this should not exists in model as model should not have validation logic...